### PR TITLE
Adding Sprite Manager and sprite sheets

### PR DESCRIPTION
**Pull request by CerberusOne**
Adding Sprite Manager class to manage all the sprite sheets.
PR also includes sprite sheets for map objects, zombies, a player, weapons and the laser shot effects.
FYI, sprite sheets will be updated again in the future.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -77,4 +77,3 @@ server: $(patsubst $(SRC)/server/$(SRCOBJS), $(OBJS), $(wildcard $(SRC)/server/*
 # Deletes the executable and all .o and .d files in the bin folder
 clean: | $(ODIR)
 	$(RM) $(EXEC) $(wildcard $(ODIR)/server*) $(wildcard $(EXEC).*) $(wildcard $(ODIR)/*.d*) $(wildcard $(ODIR)/*.o)
-
```
Comment (isaacmorneau) [inline makefile:80]: the makefile should not be changed in commits without direct instruction to do so, it at this point is rather complex and should not need further changes

File: src/Barricade.cpp [hunk 2]
```diff
@@ -63,4 +63,4 @@ void Barricade::collidingProjectile(int damage) {
 void Barricade::placeBarricade(){
     texture.setAlpha(255);
     boolPlaced=true;
-}
+}
```
Comment (isaacmorneau) [inline src/Barricade.cpp]: removal of newline is very bad practice and in fact should be added where it is missing.
for reference:
http://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file/18789
http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
https://coderwall.com/p/icxxna/don-t-forget-that-extra-line

File: src/GameStateMatch.cpp [hunk 3]
```diff
@@ -216,4 +216,4 @@ GameStateMatch::~GameStateMatch() {
     TTF_CloseFont(frameFont);
     frameFont = NULL;
 
-}
+}
```
Comment (isaacmorneau) [inline src/GameStateMatch.cpp]: keep newline

File: src/GameStateMatch.h [hunk 4]
```diff
@@ -41,11 +42,12 @@ class GameStateMatch : public GameState {
...
+
     virtual void sync();
     virtual void handle();
     virtual void update(const float& delta);
     virtual void render();
 
 };
 
-#endif
+#endif
```
Comment (isaacmorneau) [inline src/GameStateMatch.h]: keep newline

File: src/SpriteManager.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    loadSprite(ZOMBIE_BOSS);
+    loadSprite(PLAYER_MOHAWK);
+    loadSprite(WEAPONS);
+    loadSprite(LASER);
+}
+
+//Adds a texture to the _sprites map
+void SpriteManager::loadSprite(const char * fileName) {
+
+    SDL_Surface * sprite = NULL;
```
Comment (JohnAgapeyev) [inline src/SpriteManager.cpp]: Use nullptr instead of NULL

File: src/SpriteManager.cpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
+#include "SpriteManager.h"
+
+SpriteManager::SpriteManager() {
+   loadFiles();
+}
+
+SpriteManager::~SpriteManager() {
+    for (auto const& surface : _sprites) {
```
Comment (isaacmorneau) [inline src/SpriteManager.cpp]: while this works it is more likely that you meant to use `const auto& surface` instead

File: src/SpriteManager.cpp [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+//Gets a specific sprite sheet
+SDL_Surface* SpriteManager::getSurface(const char * fileName) {
+
+    SDL_Surface * sprite = _sprites[fileName];    
+
+    printf("Loading: %s\t- %s\n", fileName, SDL_GetError());
+
+    return sprite;
```
Comment (JohnAgapeyev) [inline src/SpriteManager.cpp]: Could this method not be condensed to this?

```
printf("Loading: %s\t- %s\n", fileName, SDL_GetError());
return _sprites[fileName];
```

**Review by isaacmorneau — CHANGES_REQUESTED** [review]
mostly style changes

File: src/SpriteManager.cpp [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    loadSprite(PLAYER_MOHAWK);
+    loadSprite(WEAPONS);
+    loadSprite(LASER);
+}
+
+//Adds a texture to the _sprites map
+void SpriteManager::loadSprite(const char * fileName) {
+
+    SDL_Surface * sprite = nullptr;
+    sprite = IMG_Load(fileName);
```
Comment (JohnAgapeyev) [inline src/SpriteManager.cpp:28]: Line can be condensed:

```
SDL_Surfact *sprite = IMG_Load(filename);
```

**Review by JohnAgapeyev — APPROVED** [review]

**Review by isaacmorneau — APPROVED** [review]

**Pull request merged**